### PR TITLE
Return 404 for pages with mandatory content we don't support

### PR DESCRIPTION
**Pull request by gtrufitt**
Note: needs testing, not sure if this will work as expected through Fastly.

## What does this change?

Returns 404 for pages with mandatory content that isn't supported - mainly embeds.

### Before
Returned a 500 error.

### After
Returns a 404.

image

## Why?
We were caching the error in fastly which meant that the page was not purged.

Detailed info here: https://chat.google.com/room/AAAAPTsU18A/xGbM-gEDr5o

### CODE REVIEW

**Comment by gtrufitt** [comment]
Needs article/app/renderers/RemoteRenderer.scala in Frontend updating

File: src/amp/server/render.tsx [hunk 1]
```diff
@@ -65,8 +65,11 @@ export const render = ({ body }: express.Request, res: express.Response) => {
 
         res.status(200).send(resp);
     } catch (e) {
-        // a validation error
-        if (e instanceof TypeError) {
+        if (e.message.startsWith('404')) {
```
Comment (jfsoul) [inline src/amp/server/render.tsx:68]: Could we avoid this by creating a custom error type or using the `name` property? The latter would still be a string comparison, but at least it's more obvious that it shouldn't be changed without checking elsewhere.

**Review by shtukas — APPROVED** [review]

**Comment by shtukas** [comment]
Comes together with https://github.com/guardian/frontend/pull/22728

File: src/amp/server/render.tsx (hunk 1 above)
Comment (shtukas) [reply src/amp/server/render.tsx:68]: Agree with Jon, but we can card it for later :)

**Comment by gtrufitt** [comment]
I think while this is probably still a problem, we've had exactly 0 reports so is probably not really a requirement

**Pull request closed without merging**